browser/ntp_background/ntp_custom_background_images_service_delegate.cc

```diff
@@ -57,3 +61,40 @@ std::string NTPCustomBackgroundImagesServiceDelegate::GetColor() const {
 bool NTPCustomBackgroundImagesServiceDelegate::ShouldUseRandomValue() const {
   return NTPBackgroundPrefs(profile_->GetPrefs()).ShouldUseRandomValue();
 }
+
+bool NTPCustomBackgroundImagesServiceDelegate::HasPreferredBraveBackground()
+    const {
+  const auto pref = NTPBackgroundPrefs(profile_->GetPrefs());
+  return pref.IsBraveType() && !pref.ShouldUseRandomValue() &&
+         absl::holds_alternative<GURL>(pref.GetSelectedValue());
+}
+
+base::Value::Dict
+NTPCustomBackgroundImagesServiceDelegate::GetPreferredBraveBackground() const {
+  DCHECK(HasPreferredBraveBackground());
+
+  const auto pref = NTPBackgroundPrefs(profile_->GetPrefs());
+  const auto selected_value = pref.GetSelectedValue();
+  const auto image_url = absl::get<GURL>(selected_value);
+
+  const auto* service =
+      g_brave_browser_process->ntp_background_images_service();
+  DCHECK(service);
+
+  auto* image_data = service->GetBackgroundImagesData();
+  if (!image_data) {
+    CHECK_IS_TEST();
+    return {};
+  }
+
+  auto iter = base::ranges::find_if(
+      image_data->backgrounds, [image_data, &image_url](const auto& data) {
+        return image_data->url_prefix +
+                   data.image_file.BaseName().AsUTF8Unsafe() ==
+               image_url.spec();
+      });
+
+  DCHECK(iter != image_data->backgrounds.end());
```

Comment:
IMO, this could happen when we update background components.
How about giving default(image at index 0) when this happens?

Reply:
Yes, in this patch the case is not considered. It's in discussion : https://github.com/brave/brave-browser/issues/15252#issuecomment-1153518799

Reply:
Changed to fall back to random image :)